spec/controllers/api/v2/sections_controller_spec.rb

```diff
@@ -1,128 +1,145 @@
-      get :show, params: { id: section.position + 1 }, format: :json
+    context 'when record is not present' do
+      it 'returns not found if record was not found' do
+        get :show, params: { id: section.position + 1 }, format: :json
 
-      expect(response.status).to eq 404
+        expect(response.status).to eq 404
+      end
     end
   end
-end
 
-RSpec.describe Api::V2::SectionsController, 'GET #index' do
-  let!(:chapter1) { create :chapter, :with_section }
-  let!(:chapter2) { create :chapter, :with_section }
-  let(:section1)  { chapter1.section }
-  let(:section2)  { chapter2.section }
-  let!(:section_note) { create :section_note, section_id: section1.id }
+  # GET /api/v2/sections
+  describe '#index' do
+    let!(:section1)  { create(:chapter, :with_section).section }
+    let!(:section2)  { create(:chapter, :with_section).section }
+    let!(:section_note) { create :section_note, section_id: section1.id }
 
-  let(:pattern) do
-    {
-      data: [
-        {
-          id: section1.id.to_s,
-          type: 'section',
-          attributes: {
-            id: section1.id,
-            position: section1.position,
-            title: section1.title,
-            numeral: section1.numeral,
-            chapter_from: section1.chapter_from,
-            chapter_to: section1.chapter_to,
+    let(:pattern) do
+      {
+        data: [
+          {
+            id: section1.id.to_s,
+            type: 'section',
+            attributes: {
+              id: section1.id,
+              position: section1.position,
+              title: section1.title,
+              numeral: section1.numeral,
+              chapter_from: section1.chapter_from,
+              chapter_to: section1.chapter_to,
+            },
           },
-        },
-        {
-          id: section2.id.to_s,
-          type: 'section',
-          attributes: {
-            id: section2.id,
-            position: section2.position,
-            title: section2.title,
-            numeral: section2.numeral,
-            chapter_from: section2.chapter_from,
-            chapter_to: section2.chapter_to,
+          {
+            id: section2.id.to_s,
+            type: 'section',
+            attributes: {
+              id: section2.id,
+              position: section2.position,
+              title: section2.title,
+              numeral: section2.numeral,
+              chapter_from: section2.chapter_from,
+              chapter_to: section2.chapter_to,
+            },
           },
-        },
-      ],
-    }
+        ],
+      }
+    end
+
+    it 'returns rendered records' do
+      get :index, format: :json
+
+      expect(response.body).to match_json_expression pattern
+    end
   end
 
-  it 'returns rendered records' do
-    get :index, format: :json
+  # '/api/v2/sections/:id/chapters'
+  describe '#chapters' do
+    subject(:do_request) do
+      get(:chapters, params: { id: section.id }, format: 'csv')
```

Comment:
I  tried, but I could not reuse your test "a successful csv response", 
because I could not use a path with a resource that was not yet created: 
'/api/v2/sections/<ID>/chapters'
The internal `before` block is executed after the path was actually used.

Anyway, I preferred to keep the test simple with a bit of duplication instead of doing strange things using lazy evaluation.

Reply:
If you look at the other examples, we pass a before callback inside of a block we pass to the shared example to achieve this

Reply:
https://github.com/trade-tariff/trade-tariff-backend/blob/main/spec/requests/api/v2/chapters_controller_spec.rb#L3-L8

Reply:
describe 'GET #index' do
    let(:section) { create(:section, :with_chapters) } }

    it_behaves_like 'a successful csv response', "/sections/{section.id}/chapters"
  end

Reply:
Also, we probably want a request spec for consistency and current recommended practice: https://github.com/trade-tariff/trade-tariff-backend/pull/524#discussion_r848429517

Reply:
thanks @willfish I'll try it now 👍

Reply:
naah does not work 😄 ,
let's catch up tomorrow.